File: leetcode_problems/typescript/src/lc1876_substrings_of_size_three.ts
function countGoodSubstrings(s: string): number {
  let n = s.length;
  let res = 0;
  if (n < 3) {
    return 0;
  }
  for (let i = 0; i < n - 2; i++) {
    if (s[i] !== s[i + 1] && s[i] !== s[i + 2] && s[i + 1] !== s[i + 2]) {
      res++;
    }
  }
  return res;
}

File: leetcode_problems/cpp/lc1876_substrings_of_size_three.cpp
#include <string>
using std::string;

class Solution
{
public:
    int countGoodSubstrings(string s)
    {
        int n = s.size();
        int res = 0;
        if (n < 3)
        {
            return 0;
        }
        for (int i = 0; i < n - 2; i++)
        {
            res += (s[i] != s[i + 1]) && (s[i] != s[i + 2] && (s[i + 1] != s[i + 2]));
        }
        return res;
    }
};